cmd/admission-controller/mutatingwebhook.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/scylladb/go-log"
+	"github.com/scylladb/scylla-operator-autoscaler/pkg/api/v1alpha1"
+	scyllav1 "github.com/scylladb/scylla-operator/pkg/api/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// recommendationApplier overwrites ScyllaCluster spec with recomendations given by Recommender (if available)
+type recommendationApplier struct {
+	Client  client.Client
+	decoder *admission.Decoder
+
+	c      client.Client
+	logger log.Logger
+}
+
+func getDataCenterRecommendations(sca *v1alpha1.ScyllaClusterAutoscaler) []v1alpha1.DataCenterRecommendations {
+	if sca.Status.Recommendations == nil {
+		return nil
+	}
+	dcRecs := sca.Status.Recommendations.DataCenterRecommendations
+	if len(dcRecs) == 0 {
+		return nil
+	}
+	return dcRecs
+}
+
+func getRackRecommendations(dataCenterName string,
+	dcRecs []v1alpha1.DataCenterRecommendations) []v1alpha1.RackRecommendations {
+	for idx := range dcRecs {
+		if dcRecs[idx].Name == dataCenterName {
+			return dcRecs[idx].RackRecommendations
+		}
+	}
+	return nil
+}
+
+func findRack(rackName string, racks []scyllav1.RackSpec) *scyllav1.RackSpec {
+	for idx := range racks {
+		if rackName == racks[idx].Name {
+			return &racks[idx]
+		}
+	}
+	return nil
+}
+
+func mutateCluster(ctx context.Context, logger log.Logger, cluster *scyllav1.ScyllaCluster, c client.Client) error {
+	logger.Info(ctx, "Starting mutation of ScyllaCluster")
+	scas := &v1alpha1.ScyllaClusterAutoscalerList{}
+	if err := c.List(ctx, scas); err != nil {
+		return fmt.Errorf("Failed to get SCAs: %s", err)
+	}
+
+	logger.Debug(ctx, "SCAs fetched", "num", len(scas.Items))
+
+	for idx := range scas.Items {
+		sca := &scas.Items[idx]
+
+		if *sca.Spec.UpdatePolicy.UpdateMode == v1alpha1.UpdateModeOff {
+			logger.Debug(ctx, "Autoscaler has 'off' scaling policy, skipping", "autoscaler", sca.ObjectMeta.Name)
+			continue
+		}
+
+		targetRef := sca.Spec.TargetRef
+		referencedCluster := &scyllav1.ScyllaCluster{}
+		var err = c.Get(ctx, client.ObjectKey{
+			Namespace: targetRef.Namespace,
+			Name:      targetRef.Name,
+		}, cluster)
+		if err != nil {
+			return fmt.Errorf("Fetch referenced ScyllaCluster: %s", err)
+		}
```

Comment:
It's pointless to fetch it here. You have cluster under mutation in function parameters, you can simply compare targetRef name and namespace with it without fetching.